### PR TITLE
Protobuf format containing the "raw" measurements from the sensor

### PR DESCRIPTION
**Pull request by erikfinnman**

### CODE REVIEW

File: model/raw.proto
Comment (felixeriksson) [inline model/raw.proto:1]: I think the format is good! I would like to have more information about what `range` is, easily accessible for integrators here in this document. I would suggest including the formula for converting it to units of g in a comment: 
`acceleration_g = 2 * range * acceleration_<x/y/z> / 2^16`

I also realize that we don't have any info about the units in the document, also in the old format. What would you think about adding a unit suffix to all fields that have physical units, i.e. renaming `acceleration_x` to `acceleration_x_g` in the old `RawAccelerationData`, `sample_rate` to `sample_rate_hz` osv?

**Review by felixeriksson — CHANGES_REQUESTED** [review]

File: model/raw.proto
Comment (erikfinnman) [reply model/raw.proto:1]: I have added the formula as documentation and renamed the fields.

File: model/raw.proto [hunk 1]
```diff
@@ -5,14 +5,41 @@ package model;
...
+Formula for conversion:
+
+acceleration_g = 2 * range * acceleration_<x/y/z> / 2^16
+ */
+message RawAccelerationDataCompact {
+  float sample_rate_hz = 1;
+  uint32 sample_count = 2;
+  uint32 range = 3;
+  // Acceleration values contain 2 * sample_count bytes
+  bytes acceleration_x = 4;
```
Comment (abxy) [inline model/raw.proto:32]: I suggest making these explicitly optional. An empty array or bytestring is not the same as the absence of one.

Comment (felixeriksson) [reply model/raw.proto:32]: I'm not against making them optional, but do you see a possible use in being able to differentiate empty arrays from non-existent ones?

File: model/raw.proto [hunk 2]
```diff
@@ -5,14 +5,41 @@ package model;
...
+  repeated float acceleration_z_g = 5;
+}
+
+/*
+Format containing the unscaled data from the sensor.
+
+This is a more compact format since protobuf has no built-in
+type for half-precision (16-bit) floats.
+
+The data is in little-endian.
```
Comment (felixeriksson) [inline model/raw.proto:21]:
```suggestion
The acceleration arrays are represented as packed arrays of 16-bit little endian signed integers (two's complement).
```

**Pull request merged**